branchpro/apps/simulation_dash_app.py

```diff
@@ -110,14 +110,28 @@ def update_sliders(*args):
 
 @app.app.callback(
         Output('myfig', 'figure'),
-        [Input(s, 'value') for s in sliders])
-def update_simulation(*args):
+        [Input(s, 'value') for s in sliders],
+        Input('sim-button', 'n_clicks'),
+        )
+def manage_simulation(*args):
     """
-    Simulates the model for the current slider values and updates the
+    Simulates the model for the current slider values or adds a new
+    simulation for the current slider values and updates the
     plot in the figure.
     """
-    parameters = args
-    fig = app.update_simulation(*parameters)
+    ctx = dash.callback_context
+    source = ctx.triggered[0]['prop_id'].split('.')[0]
+    if source == 'sim-button':
+        fig = app.add_simulation()
+        for i in range(len(app.plot.figure['data'])-2):
+            app.plot.figure['data'][i+1]['line'].color = 'Grey'
+    elif source in sliders:
```

Comment:
Currently it adds a separate legend entry for each simulation. What do you think of just having one legend entry for the most recent simulation? If that seems sensible, you could try something like this here:
```
app.plot.figure['data'][i+1]['showlegend'] = False
```